python/Ganga/Runtime/bootstrap.py

```diff
@@ -1279,7 +1279,8 @@ def launch_IPython(local_ns, args, error_handler, ganga_prompt):
         ipshell.confirm_exit = config['confirm_exit']
 
         # Launch embedded shell
-        ipshell(local_ns=local_ns)
+        from Ganga import GPI
+        ipshell(local_ns=local_ns, module=Ganga.GPI)
```

Comment:
It more-or-less is but not quite. It's seeded from bootstap.py's `__main__.__dict__` which, due to an import in `bin/ganga`, contains the GPI but also includes the local namespace of bootstrap.py including things like `new_version_format_to_old`, `GangaProgram` and `manualExportToGPI`. This is also the reason that doing `from __future__ import print_function` in bootstrap.py was affecting the running IPython session :confused:.

However, indeed `local_ns` *does* contain the GPI (even if there's some other things thrown in there as well). The problem we were seeing is `execfile`ing a script like:
```python
def test():
    j = Job()
test()
```
was giving `global name 'Job' is not defined`. This is because the GPI was being put into the *local* namespace of the `execfile` which is not then available inside the function `test`. This PR puts the GPI inside the *global* namespace as well so that the script can run.

Exactly whether the GPI should be in globals or locals, I'm not sure but historically we had it in both I believe.